medical-portal/src/API/RSBC.DMF.MedicalPortal.API/Controllers/DocumentController.cs

```diff
@@ -130,6 +142,84 @@ public async Task<ActionResult> GetAllDocuments([FromRoute] string driverId)
             }
         }
 
+        /// <summary>
+        /// Upload Document Content
+        /// </summary>
+        /// <param name="documentId"></param>
+        /// <returns></returns>
+        [HttpPost("upload")]     
+        [ProducesResponseType(typeof(OkResult), 200)]
+        [ProducesResponseType(401)]
+        [ProducesResponseType(500)]
+        [ActionName(nameof(UploadDriverDocument))]
+        // NOTE an operation filter was started but the generated code was not completed
+        // to save time, we opted to just write this api call manually on the frontend
+        [ApiExplorerSettings(IgnoreApi = true)]
+
+        public async Task<IActionResult> UploadDriverDocument([FromForm] IFormFile file, [FromForm] int documentSubTypeId, [FromForm] string driverId)
+        {
+            // sanity check paramters
+            if (string.IsNullOrEmpty(file?.FileName) || file.Length == 0)
+            {
+                return BadRequest();
+            }
+
+            // validate document sub type
+            var documentIdRequest = new DocumentIdRequest();
+            documentIdRequest.Id = documentSubTypeId;
+            var documentSubTypeGuidReply = _documentManagerClient.GetDocumentSubTypeGuid(documentIdRequest);
+            if (documentSubTypeGuidReply.ResultStatus != Rsbc.Dmf.CaseManagement.Service.ResultStatus.Success)
+            {
+                _logger.LogError("ERROR getting document sub type id.\n {0}", documentSubTypeGuidReply.ErrorDetail);
+                return StatusCode(500, documentSubTypeGuidReply.ErrorDetail);
+            }
+
+            var profile = await _userService.GetCurrentUserContext();
+
+            // read file stream into byte array
+            var fileStream = new MemoryStream((int)file.Length);
+            file.CopyTo(fileStream);
+            var bytes = fileStream.ToArray();
+
+            // verify mimetype from byte array
+            var mimeTypes = new MimeTypes();
+            var mimeType = mimeTypes.GetMimeType(bytes);
+            if (!DocumentUtils.IsAllowedMimeType(mimeType.Name))
+            {
+                _logger.LogError($"ERROR in uploading file due to invalid mime type {mimeType?.Name}");
+                return BadRequest("Invalid file type.");
+            }
+
+            // add the document
+            var request = new UploadFileRequest()
+            {
+                ContentType = file.ContentType,
+                Data = ByteString.CopyFrom(bytes),
+                EntityName = "dfp_driver",
```

Comment:
is it a driver file? I would think it would be a document file, and this should be "bcgov_documenturl". The entityName field links the uploaded file with a dynamics entity

Reply:
The Entity will be dfp_driver as all the uploaded documents will go the driver